packages/solidity-contracts/scripts/hardhat/eventFilter.ts

```diff
@@ -0,0 +1,137 @@
+import { task } from 'hardhat/config';
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { writeFileSync } from 'fs';
+
+task(
+  'grant-role-event-filter',
+  'Filters grant role event for a specific contract to keep track of assigned roles'
+)
+  .addParam('contract', 'address of the contract')
+  .setAction(
+    async (taskArgs: any, hre: HardhatRuntimeEnvironment): Promise<void> => {
+      const provider = new hre.ethers.JsonRpcProvider(process.env.RPC_URL);
+
+      const grantRoleEvenABI = [
```

Comment:
Maybe use hre.ethers.getContractAt to get access to the typechain types

Reply:
yeah I looked into it but still got typing errors, seems like a typing bug, will use a fix similar to the one in `upgradeVerification` script